sphinx/builders/epub3.py

```diff
@@ -157,30 +169,23 @@ def build_navlist(self, nodes):
                 continue
             if node['level'] > self.config.epub_tocdepth:
                 continue
-            if node['level'] == level:
-                navlist.append(self.new_navlist(node, level))
-            elif node['level'] == level + 1:
-                navstack.append(navlist)
-                navlist = []
-                level += 1
-                if lastnode and self.config.epub_tocdup:
```

Comment:
I agree with this is strange. But I can't determine this is needless or not. I wonder is deprecation period needed?

Note: The `epub_tocdup` option is introduced at 7cbe0588. And it is appeared since first version of epub builder (Sphinx-1.0).

BTW, now epub3 builder is very improved with all your help. I feel it is not unstable version. So it's time to replace the default epub builder with epub3.